### PR TITLE
Feature/init setup

### PR DESCRIPTION
**Pull request by naveen-variant**
Creating pull request to merge to master.

### CODE REVIEW

**Comment by naveen-variant** [comment]
Adding initial code changes.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+AWS_ROLE_ARN
+IMAGE_VERSION
+```
+
+## Output Variables
+
+Following output variables are set by default in addition to `gitversion`
+
+```text
+image-version
```
Comment (vibindaniel) [inline README.md:22]: Can you add tables here just like you added in `lazy-action-dotnet`? I liked them.

**Review by vibindaniel — CHANGES_REQUESTED** [review]
Great. Everything is coming good.

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: "Setup Action"
+description: "Sets up all the required env variables to be used by others"
+outputs:
+  image-version:
```
Comment (vibindaniel) [inline action.yml:4]: change this to `image_version`

**Review by vibindaniel — CHANGES_REQUESTED** [review]

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+## Reference: https://github.com/GitTools/actions/
+curl -sL https://raw.githubusercontent.com/GitTools/actions/main/dist/github/gitversion/execute/bundle.js -o gitversion-execute.js
+node gitversion-execute.js
+rm gitversion-execute.js
+
+echo "::add-mask::$SONAR_TOKEN"
+echo "SONAR_TOKEN=$SONAR_TOKEN" >> $GITHUB_ENV
+echo "SONAR_ORG=$SONAR_ORG" >> $GITHUB_ENV
+
+echo "AWS_ROLE_ARN=$AWS_ROLE_ARN" >> $GITHUB_ENV
```
Comment (vibindaniel) [inline entrypoint.sh:14]: Can you also add Octopus_server and Octopus_api_key?

**Review by vibindaniel — CHANGES_REQUESTED** [review]

File: action.yml (hunk 2 above)
Comment (naveen-variant) [reply action.yml:4]: Done

File: entrypoint.sh (hunk 3 above)
Comment (naveen-variant) [reply entrypoint.sh]: Added

File: README.md [hunk 4]
```diff
@@ -1,2 +1,23 @@
...
+# Lazy Action Setup
+
 Sets environment variables that will be used by other Lazy Actions
+
+## Environment Variables
+
+Following environment variables are set by default in addition to `gitversion`
+
+| Parameter                    | Description                                           |
+| ---------------------------- | ----------------------------------------------------- |
```
Comment (vibindaniel) [inline README.md:10]: You don't have a sentence in description. So please remove the periods (.).

File: entrypoint.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+## Reference: https://github.com/GitTools/actions/
+curl -sL https://raw.githubusercontent.com/GitTools/actions/main/dist/github/gitversion/execute/bundle.js -o gitversion-execute.js
+node gitversion-execute.js
+rm gitversion-execute.js
+
+echo "::add-mask::$SONAR_TOKEN"
+echo "::add-mask::$OCTOPUS_CLI_API_KEY"
+{ echo "SONAR_TOKEN=$SONAR_TOKEN"; echo "SONAR_ORG=$SONAR_ORG"; echo "OCTOPUS_CLI_SERVER=$OCTOPUS_CLI_SERVER"; echo "OCTOPUS_CLI_API_KEY=$OCTOPUS_CLI_API_KEY";} >> "$GITHUB_ENV"
+echo "AWS_ROLE_ARN=$AWS_ROLE_ARN" >> "$GITHUB_ENV"
```
Comment (vibindaniel) [inline entrypoint.sh:13]: Add an empty line

File: README.md [hunk 6]
```diff
@@ -1,2 +1,23 @@
...
+| `OCTOPUS_CLI_SERVER`         | octopus cli server.                                   |
+| `OCTOPUS_CLI_API_KEY`        | octopus cli api key .                                 |
+
+## Output Variables
+
+Following output variables are set by default in addition to `gitversion`
+
+| Parameter                    | Description                                           |
+| ---------------------------- | ----------------------------------------------------- |
+| `image_version`              | To fetch image version                                |
```
Comment (vibindaniel) [inline README.md:23]: image_version does not fetch image version. Please explain it clearly.

**Review by vibindaniel — CHANGES_REQUESTED** [review]

**Review by vibindaniel — APPROVED** [review]

**Pull request merged**